project/ExternalCompile.scala

```diff
@@ -103,14 +107,18 @@ object ExternalCompile {
             doCompile(sourcesArgs)
           }
 
-          // Output is all files in classesDirectory
-          (classesDirectory ** AllPassFilter).get.toSet
+          // Output is all files in outputDirectory
+          (outputDirectory ** AllPassFilter).get.toSet
         }
 
-        cachedCompile((sources ++ allMyDependencies).toSet)
+        val output = cachedCompile((sources ++ allMyDependencies).toSet)
+        val mapper = Path.rebase(outputDirectory, classesDirectory)
+        val mappings = output.flatMap(f => mapper(f).map(f -> _))
+
+        Sync.sync(s.cacheStoreFactory.make("compile-copy"))(mappings)
```

Comment:
Did you consider putting everything up to and including the can to `Sync.sync` inside the `cachedCompile`? It seems that this should be correct as well, and avoid useless work in `sync` when nothing's changed.

Reply:
Indeed. Fixed (even simplified some stuff).